students/scott_bromley/lesson08/circle.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+from math import sqrt, pi, pow
+
+
+class Circle(object):
+    """
+    Circle takes a radius or diameter and supports a variety of functions
+    """
+    def __init__(self, the_radius):
+        self.radius = round(float(the_radius), 2)
+
+    def __call__(self, radius, area):
+        return self._radius, self._area
+
+    @property
+    def radius(self):
+        """
+        getter for radius
+        :return: internal radius
+        """
+        return self._radius
+
+    @radius.setter
+    def radius(self, value):
+        """
+        setter for radius
+        :param value: radius
+        :return:
+        """
+        if value <= 0:
+            raise ValueError("radius must be greater than zero")
+        self._radius = value
+
+    @property
+    def diameter(self):
+        """
+        getter for diameter
+        :return: diameter of circle
+        """
+        return 2 * self.radius
+
+    @diameter.setter
+    def diameter(self, value):
+        """
+        setter for diameter; syncs radius value
+        :param value: diameter
+        :return:
+        """
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        """
+        calculate area of circle: a = pi * (r^2)
+        :return: area of circle
+        """
+        return round(pi * pow(self.radius, 2), 2)
+
+    @classmethod
+    def from_diameter(cls, value=None):
+        """
+        Circle alternate constructor
+        :param value: diameter
+        """
+        return cls(value / 2)
+
+    def __add__(self, other):
+        return Circle(self.radius + other.radius)
+
+    def __iadd__(self, other):
+        if isinstance(self, int) & isinstance(other, Circle):
+            return self.__add__(other.radius)
+        if isinstance(self, Circle) & isinstance(other, int):
+            return self.__add__(other.radius)
+        if isinstance(self, Circle) & isinstance(other, Circle):
+            return self.__add__(other.radius)
```

Comment:
general comment: you should use `and` for comparison

not sure why the if checks when you always return the same thing, but this method should look like this:
```
self.radius += other.radius
return self
```